crates/gpui/src/platform/linux/x11/client.rs

```diff
@@ -951,6 +954,8 @@ impl X11Client {
                     let click_elapsed = state.last_click.elapsed();
 
                     if click_elapsed < DOUBLE_CLICK_INTERVAL
+                        && state.last_mouse_button.is_some()
+                        && state.last_mouse_button.unwrap() == button
```

Comment:
I think this is better, that way we can avoid an unwrap.
```suggestion
                        && state.last_mouse_button.is_some_and(|prev_button| prev_button == button)
```

Reply:
Thanks for the suggestion. I updated the branch accordingly.